From eb8de6563b8e809149d0673e493d2261ca294b27 Mon Sep 17 00:00:00 2001
From: bkuhls <bkuhls@users.noreply.github.com>
Date: Sun, 26 Nov 2023 15:09:21 +0000
Subject: [PATCH] Bug 5328: Fix ESI build with libxml2 v2.12.0 (#1600)

    Libxml2Parser.cc:147:40: error: invalid conversion from
    'const xmlError*' to 'xmlErrorPtr' {aka 'xmlError*'} [-fpermissive]

libxml2 recently made xmlGetLastError() return a constant object.
---
 src/esi/Libxml2Parser.cc | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/esi/Libxml2Parser.cc b/src/esi/Libxml2Parser.cc
index f0375288511..236a37433cd 100644
--- a/src/esi/Libxml2Parser.cc
+++ b/src/esi/Libxml2Parser.cc
@@ -144,7 +144,7 @@ ESILibxml2Parser::lineNumber() const
 char const *
 ESILibxml2Parser::errorString() const
 {
-    xmlErrorPtr error = xmlGetLastError();
+    const auto error = xmlGetLastError();
 
     if (error == nullptr)
         return nullptr;
